hack/ci/setup-kubermatic-in-kind.sh

```diff
@@ -58,126 +53,19 @@ export KUBERMATIC_OIDC_PASSWORD="password"
-source <(k completion bash )
-source <(k completion bash | sed s/kubectl/k/g)
-EOF
-
 # The alias makes it easier to access the port-forwarded Dex inside the Kind cluster;
 # the token issuer cannot be localhost:5556, because pods inside the cluster would not
-# find Dex anymore.
-echodate "Setting dex.oauth alias in /etc/hosts"
-# The container runtime allows us to change the content but not to change the inode
-# which is what sed -i does, so write to a tempfile and write the tempfiles content back.
-temp_hosts="$(mktemp)"
-sed 's/localhost/localhost dex.oauth/' /etc/hosts > $temp_hosts
-cat $temp_hosts > /etc/hosts
-echodate "Set dex.oauth alias in /etc/hosts"
-
-# Create kind cluster
-TEST_NAME="Create kind cluster"
-echodate "Creating the kind cluster"
-export KUBECONFIG=~/.kube/config
-
-beforeKindCreate=$(nowms)
-nodeVersion=v1.18.2
-kind create cluster --name ${SEED_NAME} --image=kindest/node:$nodeVersion
-pushElapsed kind_cluster_create_duration_milliseconds $beforeKindCreate "node_version=\"$nodeVersion\""
-
-TEST_NAME="Pre-warm Go build cache"
-echodate "Attempting to pre-warm Go build cache"
-
-beforeGocache=$(nowms)
-make download-gocache
-pushElapsed gocache_download_duration_milliseconds $beforeGocache
-
-# Start cluster exposer, which will expose services from within kind as
-# a NodePort service on the host
-echodate "Starting cluster exposer"
-
-CGO_ENABLED=0 go build --tags "$KUBERMATIC_EDITION" -v -o /tmp/clusterexposer ./pkg/test/clusterexposer/cmd
-CGO_ENABLED=0 /tmp/clusterexposer \
-  --kubeconfig-inner "$KUBECONFIG" \
-  --kubeconfig-outer "/etc/kubeconfig/kubeconfig" \
-  --build-id "$PROW_JOB_ID" &> /var/log/clusterexposer.log &
-
-function print_cluster_exposer_logs {
-  if [[ $? -ne 0 ]]; then
-    # Tolerate errors and just continue
-    set +e
-    echodate "Printing cluster exposer logs"
-    cat /var/log/clusterexposer.log
-    echodate "Done printing cluster exposer logs"
-    set -e
-  fi
-}
-appendTrap print_cluster_exposer_logs EXIT
-
-TEST_NAME="Wait for cluster exposer"
-echodate "Waiting for cluster exposer to be running"
-
-retry 5 curl -s --fail http://127.0.0.1:2047/metrics -o /dev/null
-echodate "Cluster exposer is running"
-
-echodate "Setting up iptables rules for to make nodeports available"
-KIND_NETWORK_IF=$(ip -br addr | grep -- 'br-' | cut -d' ' -f1)
-
-iptables -t nat -A PREROUTING -i eth0 -p tcp -m multiport --dports=30000:33000 -j DNAT --to-destination 172.18.0.2
-# By default all traffic gets dropped unless specified (tested with docker
-# server 18.09.1)
-iptables -t filter -I DOCKER-USER -d 172.18.0.2/32 ! -i $KIND_NETWORK_IF -o $KIND_NETWORK_IF -p tcp -m multiport --dports=30000:33000 -j ACCEPT
-
-# Docker sets up a MASQUERADE rule for postrouting, so nothing to do for us
-echodate "Successfully set up iptables rules for nodeports"
+# find Dex anymore. As this script can be run multiple times in the same CI job,
+# we must make sure to only add the alias once.
```

Comment:
Currently this is not true, the script is not run multiple times. But it soon will be.